test/e2e/scale_test.go

```diff
@@ -0,0 +1,161 @@
+			var names test.ResourceNames
+			names.Service = test.AppendRandomString(fmt.Sprintf("scale-%05d-%03d-", scale, i), logger)
+
+			svc, err := test.CreateLatestServiceWithResources(logger, clients, names, imagePath)
+			if err != nil {
+				t.Fatalf("Failed to create Service: %v", err)
+			}
+			names.Route = serviceresourcenames.Route(svc)
+			names.Config = serviceresourcenames.Configuration(svc)
+
+			// Send it to our cleanup logic (below)
+			cleanupCh <- names
+
+			logger.Infof("Wait for %s to become ready.", names.Service)
+			if err := test.WaitForServiceState(clients.ServingClient, names.Service, test.IsServiceReady, "ServiceIsReady"); err != nil {
+				return err
+			}
+
+			var domain string
+			err = test.WaitForServiceState(clients.ServingClient, names.Service, func(s *v1alpha1.Service) (bool, error) {
+				if s.Status.Domain != "" {
+					domain = s.Status.Domain
+					return true, nil
+				}
+				return false, nil
+			}, "ServiceUpdatedWithDomain")
+			if err != nil {
+				t.Fatalf("Service %s was not updated with a domain: %v", names.Service, err)
+			}
+
+			_, err = pkgTest.WaitForEndpointState(
+				clients.KubeClient,
+				logger,
+				domain,
+				pkgTest.Retrying(pkgTest.EventuallyMatchesBody(helloWorldExpectedOutput), http.StatusNotFound),
+				"WaitForEndpointToServeText",
+				test.ServingFlags.ResolvableDomain)
+			if err != nil {
+				t.Fatalf("The endpoint for Service %s at domain %s didn't serve the expected text \"%s\": %v", names.Service, domain, helloWorldExpectedOutput, err)
+			}
+			domainCh <- domain
+
+			logger.Infof("%s is ready.", names.Service)
+			return nil
+		})
+	}
+
+	go func() {
+		if err := deployGrp.Wait(); err != nil {
+			t.Fatalf("Error waiting for endpoints to become ready: %v", err)
+		}
+		close(domainCh)
+	}()
+
+	for {
+		select {
+		case names := <-cleanupCh:
+			test.CleanupOnInterrupt(func() { TearDown(clients, names, logger) }, logger)
+			defer TearDown(clients, names, logger)
+
+		case domain, ok := <-domainCh:
+			if !ok {
+				logger.Info("All services were created successfully.")
+				return
+			}
+
+			// Start probing the domain until the test is complete.
+			probeCh := test.RunRouteProber(logger, clients, domain)
+			defer func(probeCh <-chan error) {
+				if err := test.GetRouteProberError(probeCh, logger); err != nil {
+					t.Fatalf("Route prober failed with error %s", err)
+				}
+			}(probeCh)
+
+		case <-time.After(duration):
+			t.Fatalf("Timed out waiting for %d services to become ready", scale)
+		}
+	}
+}
+
+// While redundant, we run two versions of this by default:
+// 1. TestScaleTo10: a developer smoke test that's useful when changing this to assess whether
+//   things have gone horribly wrong.  This should take about 12-20 seconds total.
+// 2. TestScaleTo50: a more proper execution of the test, which verifies a slightly more
+//   interesting burst of deployments, but low enough to complete in a reasonable window.
+
+func TestScaleTo10(t *testing.T) {
```

Comment:
t.Parallel() here and below?

Reply:
We don't want these to run in parallel, we'll have cluster capacity problems.